### ISSUE TITLE
Open Python Interactive Programmatically

### ISSUE THREAD

**Issue by shaggygi** [Microsoft/PTVS#389]
I would like to create a project template that will open the Python Interactive Window and add a few lines of code (similar to how a user would manually).

1.) How do you perform this in C# programmatically?

2.) In addition, what if I had custom commands in my project template.  How would you get a handle to the Python Interactive Window (assuming it is already opened) to add more lines of code to it?

Thanks

**Comment by zooba** [comment]
Currently, this is possible with the interactive window that's part of PTVS (or NTVS) for VS 2013, but the new interactive window in VS 2015 does not have obvious support for it unless you're creating the entire window yourself.

You can look at our test projects for PTVS to see how we do it for VS 2013, but we're still figuring out how best to do it for VS 2015 and chances are it will only be appropriate for testing.

Once you get an `IReplWindow` or `IInteractiveWindow` interface then you can insert or submit code into the interactive window through those. (Getting the `IInteractiveWindow` interface is currently the tricky part.)

**Comment by shaggygi** [comment]
Thanks for the response.  Is the IInteractiveWindow interface part of what is coming with VS 2015 RTM or Roslyn v1 RTM or something else?

**Comment by zooba** [comment]
It's in VS 2015 RC, but we still need to get a reference assembly added to the VSSDK so it can be used. (Currently we're using the DLLs from PrivateAssemblies, which we're not meant to be doing...)

**Comment by shaggygi** [comment]
OK.  Thanks.  I'll keep an eye out.  Hopefully (and if you can remember between busy projects), you could send out a quick update on this thread to give a heads up :)

**Comment by zooba** [comment]
So the way we're doing this in VS 2015 (only for testing) is to keep a private list of open windows as they open (in PTVS), and using the regular command from tests to trigger them. This isn't exposed in any way publicly, so there's currently no way to get access to already open windows without walking the WPF tree (highly not recommended - this is very brittle).

It probably won't show up until the next major version of VS, but since the interactive window is now standard we've got more teams trying to use it, so there will be API changes. Chances are this will be come easier (specifically, I'm keen to have some sort of extensibility API to open interactive windows and pass in some code/file/module to execute). Will have to wait and see how this turns out, but it is something that the broader VS team is interested in.

Closing this now, as there's nothing more I can add until other teams start talking about what they've added to the next VS. Keep an eye on the VS blog - it'll probably show up there first.
